### PR TITLE
CPDTP-428: Csv participant declarations

### PR DESCRIPTION
**Pull request by kicferk1**
### Context
We want to support CSV formats to make ourselves as flexible to the providers importing data as possible.
https://dfedigital.atlassian.net/browse/CPDTP-428

### Changes proposed in this pull request
Add an csv format support to existing participant declarations index endpoint.
Pull 'to_csv' method into its own module to share it between controllers.
Specify the format of declaration date in serialiser explicitly to avoid `to_s` in csv computation returning a different format.
Add documentation and tests.
Fix the githook saying `/components/schemas/XXXXXX` is a bloody AWS key. 

### Guidance to review
I imagine it will have some conflicts with https://github.com/DFE-Digital/early-careers-framework/pull/952, but they shouldn't be too terrible. 

### Testing
Added request specs for the csv index endpoint

### How can I view this in a review app?
Hit the `/api/v1/participant-declarations.csv` endpoint with the right token.

### CODE REVIEW

File: app/controllers/concerns/api_csv.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
+module ApiCsv
+  extend ActiveSupport::Concern
```
Comment (blackrat) [inline app/controllers/concerns/api_csv.rb:4]: Don't need to extend Concern unless you're using `included` `prepended` or one of the other methods that it adds. This is just a plain ol' grouping module.

File: swagger/v1/component_schemas/MultipleParticipantDeclarationsCsvResponse.yml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+description: "A list of participant declarations in the Comma Separated Value (CSV) format"
+type: string
+format: base64
```
Comment (blackrat) [inline swagger/v1/component_schemas/MultipleParticipantDeclarationsCsvResponse.yml:3]: Is it?

File: app/controllers/concerns/api_csv.rb (hunk 1 above)
Comment (kicferk1) [reply app/controllers/concerns/api_csv.rb:4]: Good spot :)

File: swagger/v1/component_schemas/MultipleParticipantDeclarationsCsvResponse.yml (hunk 2 above)
Comment (kicferk1) [reply swagger/v1/component_schemas/MultipleParticipantDeclarationsCsvResponse.yml:3]: No it ain't.
I need to purge that from a few other places too...

**Review by blackrat — APPROVED** [review]

**Pull request merged**